templates/common/baremetal/files/NetworkManager-mdns-hostname.yaml

```diff
@@ -0,0 +1,21 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/40-mdns-hostname"
+contents:
+  inline: |
+    #!/bin/bash
+    STATUS=$2
+    case "$STATUS" in
+        dhcp4-change|dhcp6-change|hostname)
+        logger -s "NM mdns-hostname triggered by ${2}."
+        set +e
+        t_hostname=$(hostname)
+        if [ -z "${t_hostname}" ]; then
+           t_hostname="localhost"
+        fi
+        echo "${t_hostname}">/etc/mdns/hostname.txt
```

Comment:
I think we can safely drop the `.txt` extension

Reply:
OK